test/serializers/associations_test.rb

```diff
@@ -145,6 +145,26 @@ def test_associations_custom_keys
         assert expected_association_keys.include? :writer
         assert expected_association_keys.include? :site
       end
+
+      def test_belongs_to_doesnt_load_record
+        post = Post.new({ id: 997 })
+
+        class << post
+          def blog_id
+            996
+          end
+          def blog
+            fail "should use blog_id"
+          end
+        end
+
+        post_serializer = Class.new(ActiveModel::Serializer) do
```

Comment:
I would call the class `PostSerializer` instead of `post_serializer` (although I know you took the code from somewhere else, but I think this is more 'standard').
Also, I would write it like that:
```ruby
class PostSerializer < ActiveModel::Serializer
  belongs_to :blog, embed: :ids
end
```
as I find it less confusing, but here people might disagree.